File: app/src/main/java/ru/samtakoy/core/app/utils/DateUtils.java
package ru.samtakoy.core.app.utils;

import java.util.Date;

public class DateUtils {

    public static long getTimeLong(Date date) {
        return date.getTime();
    }

    public static long getCurrentTimeLong() {
        return System.currentTimeMillis();
    }

    public static long getMillisTo(long targetUtc) {
        return targetUtc - System.currentTimeMillis();
    }

    public static long getMillisTo(Date date) {
        return getMillisTo(dateToDbSerialized(date));
    }

    public static Date getCurrentTimeDate() {
        return getDateFromLong(getCurrentTimeLong());
    }

    public static long getDateAfterCurrentLong(int millisDelta) {
        return getCurrentTimeLong() + millisDelta;
    }

    public static long dateToDbSerialized(Date d){
        return getTimeLong(d);
    }

    public static Date dateFromDbSerialized(long utcTimeMillis){
        return getDateFromLong(utcTimeMillis);
    }


    public static Date getDateFromLong(long value) {
        return new Date(value);
    }

}

File: app/src/main/kotlin/ru/samtakoy/core/app/utils/DateUtils.kt
package ru.samtakoy.core.app.utils

import java.util.Date

object DateUtils {
    fun getTimeLong(date: Date): Long {
        return date.getTime()
    }

    val currentTimeLong: Long
        get() = System.currentTimeMillis()

    fun getMillisTo(targetUtc: Long): Long {
        return targetUtc - System.currentTimeMillis()
    }

    fun getMillisTo(date: Date): Long {
        return getMillisTo(dateToDbSerialized(date))
    }

    val currentTimeDate: Date
        get() = getDateFromLong(currentTimeLong)

    fun getDateAfterCurrentLong(millisDelta: Int): Long {
        return currentTimeLong + millisDelta
    }

    fun dateToDbSerialized(d: Date): Long {
        return getTimeLong(d)
    }

    fun dateFromDbSerialized(utcTimeMillis: Long): Date {
        return getDateFromLong(utcTimeMillis)
    }

    fun getDateFromLong(value: Long): Date {
        return Date(value)
    }
}
